src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -660,16 +630,69 @@ private void LogRenameSession(RenameLogMessage.UserActionOutcome outcome, bool p
         }
 
         public void Cancel()
-            => Cancel(rollbackTemporaryEdits: true);
-
-        private void Cancel(bool rollbackTemporaryEdits)
         {
             _threadingContext.ThrowIfNotOnUIThread();
             VerifyNotDismissed();
 
-            LogRenameSession(RenameLogMessage.UserActionOutcome.Canceled, previewChanges: false);
-            Dismiss(rollbackTemporaryEdits);
-            EndRenameSession();
+            DismissUIAndRollbackEditsAndEndRenameSession(RenameLogMessage.UserActionOutcome.Canceled, previewChanges: false);
+        }
+
+        private void DismissUIAndRollbackEditsAndEndRenameSession(
+            RenameLogMessage.UserActionOutcome outcome,
+            bool previewChanges,
+            Action finalCommitAction = null)
+        {
+            // Note: this entire sequence of steps is not cancellable.  We must perform it all to get back to a correct
+            // state for all the editors the user is interacting with.
+
+            // Remove all our adornments and restore all buffer texts to their initial state.
+            DismissUIAndRollbackEdits();
+
+            // We're about to perform the final commit action.  No need to do any of our BG work to find-refs or compute conflicts.
+            _cancellationTokenSource.Cancel();
+            _conflictResolutionTaskCancellationSource.Cancel();
```

Comment:
these two cancels were in CancelAllOpenDocumentTrackingTasks.  This was the only place that called that, so this was inlined here.